### PR TITLE
feat: return bucket index of the current tick from LiquidityPerTickRange query

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: https://app.clickup.com/t/86a19jj0x

## What is the purpose of the change

This PR is an update to the full range liquidity query. It adds a new return parameter that is the bucket index corresponding to the current tick.

This functionality is necessary for the SQS router in: https://github.com/osmosis-labs/osmosis/pull/6785

By having this index, we don't have to iterate all ticks in search of the current tick. Instead, we can index into the desired bucket in constant time.

## Testing and Verifying

Added unit test and also tested on localosmosis

## Documentation and Release Note

  - [x] Does this pull request introduce a new feature or user-facing behavior changes?
  - [x] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

### CODE REVIEW

File: x/concentrated-liquidity/query.go [hunk 1]
```diff
@@ -19,52 +19,73 @@ import (
 
 // This file contains query-related helper functions for the Concentrated Liquidity module
 
-// GetTickLiquidityForFullRange returns an array of liquidity depth for all ticks existing from min tick ~ max tick.
-func (k Keeper) GetTickLiquidityForFullRange(ctx sdk.Context, poolId uint64) ([]queryproto.LiquidityDepthWithRange, error) {
+// GetTickLiquidityForFullRange returns a slice of liquidity buckets for all tick range existing from min tick ~ max tick.
```
Comment (p0mvn) [inline x/concentrated-liquidity/query.go:24]:
```suggestion
// GetTickLiquidityForFullRange returns a slice of liquidity buckets for all tick ranges existing from min tick ~ max tick.
```

File: x/concentrated-liquidity/query.go [hunk 2]
```diff
@@ -17,54 +17,77 @@ import (
...
+
 // This file contains query-related helper functions for the Concentrated Liquidity module
 
-// GetTickLiquidityForFullRange returns an array of liquidity depth for all ticks existing from min tick ~ max tick.
-func (k Keeper) GetTickLiquidityForFullRange(ctx sdk.Context, poolId uint64) ([]queryproto.LiquidityDepthWithRange, error) {
+// GetTickLiquidityForFullRange returns a slice of liquidity buckets for all tick ranges existing from min tick ~ max tick.
+// Returns index of the bucket that corresponds to the current tick.
+// For cases where there is no liqudity in the bucket but there may be liquidity to the right, the value will be -1.
+// For cases where there is no liquidity in the bucket but there may be liquidity to the left, the value will be len(liquidityDepthsForRange).
+// Otherwise, the index points to the bucket that corresponds to the current tick.
```
Comment (AlpinYukseloglu) [inline x/concentrated-liquidity/query.go:28]: It might be useful to more concretely define "bucket" vs. "tick" either here or in docs

**Review by AlpinYukseloglu — APPROVED** [review]
LGTM assuming you've already tested the changes on localosmosis (I just reviewed code).

Had one comment around more clearly defining "bucket" for external users of this query but non-blocking so approving

File: x/concentrated-liquidity/query.go (hunk 2 above)
Comment (mattverse) [reply x/concentrated-liquidity/query.go:28]: The docs defines terminologies for these AFAIR!

Comment (p0mvn) [reply x/concentrated-liquidity/query.go:28]: https://github.com/osmosis-labs/osmosis/blob/f721f73fa3afeaa16b39c9da89b25cdac4c98ee4/x/concentrated-liquidity/README.md?plain=1#L1893-L1895

**Pull request merged**